### PR TITLE
Add support for serving esm files directly from unpkg.

### PR DESCRIPTION
**Pull request by alexjeffburke**
The following changes have been made:
* app proxy()
  allow a ${v} placeholder which supports passing the version to
  unpkg and converts a "master" branch name to the "latest" version
* render
  support an "esm" type that serves the files directly
* render_code
  bypass typescript file annotation when files are type "esm"

### CODE REVIEW

**Comment by j-f1** [comment]
Why not annotate ESM files?

**Comment by alexjeffburke** [comment]
@f-j1 from what I understood the annotation machinery relies of the typescript compiler generating an AST. In the case of a pre-existing ESM build that contained deno support, my reasoning was you wouldn’t want to invoke the compiler on this.

Also, the current pretty line numbering stuff when used with very large builds was causing the browser to really struggle so I’m not sure the added weight of links wouldn’t gain much.

**Comment by j-f1** [comment]
> from what I understood the annotation machinery relies of the typescript compiler generating an AST

That’s correct. However, that doesn’t mean you need to use TypeScript to actually transpile the module. Also, TypeScript works OK on JS, especially for the jump-to-definition stuff we’re doing now

> Also, the current pretty line numbering stuff when used with very large builds was causing the browser to really struggle so I’m not sure the added weight of links wouldn’t gain much.

How about instead checking the size of the source file and skipping the pretty-rendering pipeline if it exceeds a certain value?

**Comment by alexjeffburke** [comment]
Ok - I’m more than happy to look at that, but is that enough to allow this to land?

I’d really like to get Unexpected listed in the registry and this is somewhat of a blocker.

**Comment by j-f1** [comment]
It should be, but I’m not able to land it myself — only @ry can do that AFAIK.

**Comment by ry** [comment]
This looks good to me - but I want to land #151 first.

@j-f1 - you've reviewed and approve of this patch?

**Comment by j-f1** [comment]
LGTM except for my comment above.

**Comment by alexjeffburke** [comment]
@j-f1 @ry just rebased this branch on master.

Re the pretty print, Had a look again it is working for me here and I think it's actually quite nice :) If anything I'd opt perhaps to truncate the output beyond a certain point but my feeling is that could easily be done as a follow-up - but let me know what you think is best.

**Comment by ry** [comment]
@alexjeffburke Sorry for the delay on this. Can you merge with master, please?

File: src/utils.js [hunk 1]
```diff
@@ -28,18 +28,32 @@ exports.walkAST = function walkAST(node, cb) {
  */
 exports.getEntry = function getEntry(name, branch = "master") {
   const rawEntry = DATABASE[name];
-  if (rawEntry.type === "url") {
+  if (!rawEntry) {
```
Comment (alexjeffburke) [inline src/utils.js]: Just noting that without this serving a package that does not exist is causing a bad dereference of type - this ensures a 404 is a safely served.

**Comment by alexjeffburke** [comment]
@ry have just pushed this rebased on latest master :)

**Comment by ry** [comment]
Thanks @alexjeffburke LGTM

**Pull request merged**